spec/haproxy/templates/drain_spec.rb

```diff
@@ -31,8 +31,7 @@
           fi
 
           pid="$(cat ${pidfile})"
-          pid_exists=$(ps -p ${pid} | tail -n +2)
-          if [ "${pid_exists}" = '' ]; then
+          if ![ ps -p ${pid} >/dev/null ]; then
```

Comment:
this doesn't work.
```
if ! ps -p ${pid} >/dev/null; then
```
you need to lose the brackets as this is not a test anymore. you are now executing a program and testing for its exit code.